text/0000-rehydration-by-default.md

```diff
@@ -0,0 +1,66 @@
+- Start Date: 2019-02-22
+- Relevant Team(s): Ember CLI / Fastboot
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking:
+
+# Enable Rehydration By Default
+
+## Summary
+
+Firstly, I'd like to take a brief minute to talk about what Rehydration is for the curious.  To do so I'm going to pull in some of the explanation from the commit that introduced it to the Glimmer VM:
+
+> -- What is Rehydration?
+>
+> The rehydration feature means that the Glimmer VM can take a DOM tree
+created using Server Side Rendering (SSR) and use it as the starting
+point for the append pass.
+>
+> Rehydration is optimized for server-rendered DOMs that will not need
+many changes in the client, but it is also capable of making targeted
+repairs to the DOM:
+>
+> - if a single text node's value changes between the server and the
+>  client, the text node itself is repaired.
+> - if the contents of a `{{{}}}` (or SafeString) change, only the
+>   bounds of that fragment of HTML are blown away and recreated.
+> - if a mismatch is found, only the remainder of the current element
+  is cleared.
+>
+> What this means in practice is that rehydration repairs problems
+in a relatively targeted way, and doesn't blows away more content
+than the contents of the current element when a mismatch occurs.
+>
+>Near-term work will narrow the scope of repairs further, so that
+mismatches inside of blocks only clear out the remainder of the
+content of the block.
+
+<small>(from: https://github.com/glimmerjs/glimmer-vm/commit/316805b9175e01698120b9566ec51c88d075026a)<small>
+
+Currently Fastboot rehydration is available behind a (rather lengthy) environment variable: `EXPERIMENTAL_RENDER_MODE_SERIALIZE=true`.  The Goal of this RFC is to enable this by default and explore the work necessary to accomplish that.
+
+## Motivation
+
+Much of the work done in FastBoot today is duplicated because rehydration is not enabled.  In fact, the current implementation of `ember-cli-fastboot` has an initializer ([here](https://github.com/ember-fastboot/ember-cli-fastboot/blob/master/addon/instance-initializers/clear-double-boot.js)) called `clear-double-boot` which removes all of the work done in SSR land so the Glimmer VM has a fresh start to begin creating the DOM. With Rehydration the work done on the server is not thrown away and instead re-used, so removing the `clear-double-boot` should result in faster websites.
+
```

Comment:
"faster websites" is a good motivation

I'd also mention that a downside of `clear-double-boot` is that if a user starts filling out a form before the app has completed the in-browser boot, their entries are clobbered by the `double-boot` process. This is relatively "uncommon" however we've seen it happen on cell phones when assets take an inexplicably long time to load.

Replacing the entire dom with a newly-generated dom is potentially like having the rug pulled out from under you. So "motivation" here is "not pulling the rug out from under users."

Reply:
I've updated a section around `scenarios` to point this out